sql/src/main/java/io/crate/execution/engine/collect/files/LineInput.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to CRATE Technology GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.engine.collect.files;
+
+import io.crate.execution.dsl.phases.FileUriCollectPhase;
+import io.crate.operation.collect.files.CSVLineParser;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+
+public class LineInput {
+
+    private CSVLineParser csvLineParser;
+
+    private InputType inputType;
+
+    private enum InputType {
```

Comment:
Hm I don't think so. My reason for keeping them separate was because the input type in the `LineParser` might not be the same as the `FileUriCollectPhase` input format. 

i.e. in the scenario where the file extension is `.csv`, the `FileUriCollectPhase.inputFormat` will be set to the default value (`JSON`), whereas during line parsing the `LineParser.inputType` will be set to `CSV` (given that this  is decided based on both the `FileUriCollectPhase.inputFormat` and file extension).